spec/rest/http.test.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+define(['ably', 'shared_helper'], function(Ably, helper) {
+	var rest, exports = {},
+		Defaults = Ably.Rest.Defaults;
+
+	exports.setupHttp = function(test) {
+		test.expect(1);
+		helper.setupApp(function() {
+			rest = helper.AblyRest();
+			test.ok(true, 'App created');
+			test.done();
+		});
+	}
+
+	/**
+	 * Check presence of X-Ably-Version headers in get&post requests
+	 * @spec : (RSC7a)
+	 */
+	exports.apiVersionHeader = function(test) {
+
+		//Intercept get&post methods with test
+		var get_inner = Ably.Rest.Http.get;
+		Ably.Rest.Http.get = function (rest, path, headers, params, callback) {
+			test.ok(('X-Ably-Version' in headers), 'Verify version header exists');
+			test.equal(headers['X-Ably-Version'], Defaults.apiVersion, 'Verify current version number');
+			return get_inner(rest, path, headers, params, callback);
+		};
+
+		var post_inner = Ably.Rest.Http.post;
+		Ably.Rest.Http.post = function (rest, path, headers, body, params, callback) {
+			test.ok(('X-Ably-Version' in headers), 'Verify version header exists');
+			test.equal(headers['X-Ably-Version'], Defaults.apiVersion, 'Verify current version number');
+			return post_inner(rest, path, headers, body, params, callback);
+		};
+
+		//Call all methods that use rest http calls
+		test.expect(10);
+
+		rest.auth.requestToken();
+		rest.time();
+		rest.stats();
+
+		var channel = rest.channels.get('http_test_channel');
+		channel.publish('test', 'Testing http headers');
+		channel.presence.get();
+
+		//Clean interceptors from get&post methods
+		Ably.Rest.Http.get = get_inner;
```

Comment:
This test ends (with `test.done()`) while there are requests still in progress, because it doesn't wait for callbacks/responses from the REST calls it makes. Is that an issue? I would prefer that it waits, so there is no other activity going on when later tests run.

In fact instead of that, why do you even need to call `post_inner()` and `get_inner()` ? It serves no purposes in the test.